### PR TITLE
Settings defaults to null before api returns object

### PR DESCRIPTION
**Pull request by MorganLove**
**Description**

- Trying to refactor this to be more consistent in api usage

**Testing**

- It works, but there's one test that's failing and I'm not sure how it's even related.

**Progress**

- Still puzzled by this.

### CODE REVIEW

**Comment by jonahkagan** [comment]
here's a patch that fixes that test (though there are prob more changes that need to happen as a result)

```
diff --git a/client/src/components/HomeScreen.test.tsx b/client/src/components/HomeScreen.test.tsx
index 9c9de4c..9a21bac 100644
--- a/client/src/components/HomeScreen.test.tsx
+++ b/client/src/components/HomeScreen.test.tsx
@@ -4,6 +4,7 @@ import userEvent from '@testing-library/user-event'
 import { withMockFetch, renderWithRouter } from './testUtilities'
 import App from '../App'
 import { aaApiCalls, jaApiCalls } from './MultiJurisdictionAudit/_mocks'
+import { auditSettings } from './MultiJurisdictionAudit/useSetupMenuItems/_mocks'
 
 const apiCalls = {
   unauthenticatedUser: {
@@ -78,8 +79,8 @@ const setupScreenCalls = [
   aaApiCalls.getRounds,
   aaApiCalls.getJurisdictions,
   aaApiCalls.getContests,
-  aaApiCalls.getSettings,
-  aaApiCalls.getSettings,
+  aaApiCalls.getSettings(auditSettings.blank),
+  aaApiCalls.getSettings(auditSettings.blank),
   aaApiCalls.getJurisdictionFile,
 ]
 
diff --git a/client/src/components/MultiJurisdictionAudit/_mocks.ts b/client/src/components/MultiJurisdictionAudit/_mocks.ts
index 02c74aa..fa3454d 100644
--- a/client/src/components/MultiJurisdictionAudit/_mocks.ts
+++ b/client/src/components/MultiJurisdictionAudit/_mocks.ts
@@ -6,6 +6,7 @@ import {
   talliesFile,
 } from './useSetupMenuItems/_mocks'
 import { FileProcessingStatus } from './useSetupMenuItems/getJurisdictionFileStatus'
+import { IAuditSettings } from '../../types'
 
 const manifestFormData: FormData = new FormData()
 manifestFormData.append('manifest', manifestFile, manifestFile.name)
@@ -172,10 +173,10 @@ export const aaApiCalls = {
     url: '/api/election/1/contest',
     response: contestMocks.filledTargeted,
   },
-  getSettings: {
+  getSettings: (settings: IAuditSettings) => ({
     url: '/api/election/1/settings',
-    response: auditSettings.all,
-  },
+    response: settings,
+  }),
   putSettings: {
     url: '/api/election/1/settings',
+import { IAuditSettings } from '../../types'
 
 const manifestFormData: FormData = new FormData()
 manifestFormData.append('manifest', manifestFile, manifestFile.name)
@@ -172,10 +173,10 @@ export const aaApiCalls = {
     url: '/api/election/1/contest',
     response: contestMocks.filledTargeted,
   },
-  getSettings: {
+  getSettings: (settings: IAuditSettings) => ({
     url: '/api/election/1/settings',
-    response: auditSettings.all,
-  },
+    response: settings,
+  }),
   putSettings: {
     url: '/api/election/1/settings',
     options: {
```

File: client/src/components/MultiJurisdictionAudit/useAuditSettings.ts [hunk 1]
```diff
@@ -25,15 +17,15 @@ type TNewSettings =
...
-  const getSettings = useCallback(async (): Promise<IAuditSettings> => {
+  const getSettings = useCallback(async (): Promise<IAuditSettingsPossNull> => {
     const response = await api<IAuditSettings>(
       `/election/${electionId}/settings`
     )
     if (!response) {
-      return defaultValues
+      return null
     }
     return response
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/useAuditSettings.ts:30]: this whole function body can be reduced to:

```ts
return api(`/election/${electionId}/settings`)
```

File: client/src/components/testUtilities.tsx [hunk 2]
```diff
@@ -73,7 +73,7 @@ export const renderWithRouter = (
 interface FetchRequest {
   url: string
   options?: RequestInit
-  response: object
+  response: object | null
```
Comment (jonahkagan) [inline client/src/components/testUtilities.tsx:76]: i dont think you need this change, though I might be missing something

File: client/src/components/MultiJurisdictionAudit/Progress/index.test.tsx [hunk 3]
```diff
@@ -13,7 +13,7 @@ describe('Progress screen', () => {
     const { container } = render(
       <Progress
         jurisdictions={jurisdictionMocks.oneManifest}
-        auditSettings={auditSettings.all}
+        auditSettings={auditSettings.all!}
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Progress/index.test.tsx:16]: do you still need all these exclamation points?

**Review by jonahkagan — APPROVED** [review]

File: client/src/components/MultiJurisdictionAudit/Progress/index.test.tsx (hunk 3 above)
Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Progress/index.test.tsx]: @jonahkagan Evidently not, haha. Fixed and pushed. :slightly_smiling_face:

**Pull request merged**